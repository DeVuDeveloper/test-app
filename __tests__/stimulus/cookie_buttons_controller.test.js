import { screen } from '@testing-library/dom';
import { getHTML, setHTML, startStimulus } from './_stimulus_helper';
import CookieButtonsController from '../../app/javascript/controllers/cookie_buttons_controller';

beforeEach(() => startStimulus('cookie-buttons', CookieButtonsController));

test('selects a filling and updates UI', async () => {
  await setHTML(`
    <div data-controller="cookie-buttons">
      <button class="filling-button" data-filling="chocolate">Chocolate</button>
      <button class="filling-button" data-filling="strawberry">Strawberry</button>
      <button class="filling-button" data-filling="vanilla">Vanilla</button>
      <input type="hidden" data-cookie-buttons-target="selectedFilling" value="">
    </div>
  `);

  await screen.findByText('Chocolate');
  const chocolateButton = screen.getByText('Chocolate');
  const strawberryButton = screen.getByText('Strawberry');

  expect(getHTML()).toMatchSnapshot();

  chocolateButton.click();
  await screen.findByRole('button', { name: 'Chocolate' });

  expect(getHTML()).toMatchSnapshot();

  strawberryButton.click();
  await screen.findByRole('button', { name: 'Strawberry' });

  expect(getHTML()).toMatchSnapshot(); 
});
