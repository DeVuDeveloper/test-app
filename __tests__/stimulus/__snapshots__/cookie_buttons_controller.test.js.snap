// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selects a filling and updates UI 1`] = `
"<div data-controller="cookie-buttons">
      <button class="filling-button" data-filling="chocolate">Chocolate</button>
      <button class="filling-button" data-filling="strawberry">Strawberry</button>
      <button class="filling-button" data-filling="vanilla">Vanilla</button>
      <input type="hidden" data-cookie-buttons-target="selectedFilling" value="">
    </div>"
`;

exports[`selects a filling and updates UI 2`] = `
"<div data-controller="cookie-buttons">
      <button class="filling-button" data-filling="chocolate">Chocolate</button>
      <button class="filling-button" data-filling="strawberry">Strawberry</button>
      <button class="filling-button" data-filling="vanilla">Vanilla</button>
      <input type="hidden" data-cookie-buttons-target="selectedFilling" value="">
    </div>"
`;

exports[`selects a filling and updates UI 3`] = `
"<div data-controller="cookie-buttons">
      <button class="filling-button" data-filling="chocolate">Chocolate</button>
      <button class="filling-button" data-filling="strawberry">Strawberry</button>
      <button class="filling-button" data-filling="vanilla">Vanilla</button>
      <input type="hidden" data-cookie-buttons-target="selectedFilling" value="">
    </div>"
`;
